crates/db_views/src/post_view.rs

```diff
@@ -346,47 +346,30 @@ fn queries<'a>() -> Queries<
       query = query.filter(post_aggregates::creator_id.eq(creator_id));
     }
 
-    if let Some(listing_type) = options.listing_type {
-      if let Some(person_id) = options.local_user.person_id() {
-        let is_subscribed = exists(
-          community_follower::table.filter(
-            post_aggregates::community_id
-              .eq(community_follower::community_id)
-              .and(community_follower::person_id.eq(person_id)),
-          ),
-        );
-        match listing_type {
-          ListingType::Subscribed => query = query.filter(is_subscribed),
-          ListingType::Local => {
-            query = query
-              .filter(community::local.eq(true))
-              .filter(community::hidden.eq(false).or(is_subscribed));
-          }
-          ListingType::All => query = query.filter(community::hidden.eq(false).or(is_subscribed)),
-          ListingType::ModeratorView => {
-            query = query.filter(exists(
-              community_moderator::table.filter(
-                post::community_id
-                  .eq(community_moderator::community_id)
-                  .and(community_moderator::person_id.eq(person_id)),
-              ),
-            ));
-          }
-        }
+    let is_subscribed = exists(
+      community_follower::table.filter(
+        post_aggregates::community_id
+          .eq(community_follower::community_id)
+          .and(community_follower::person_id.eq(person_id_join)),
+      ),
+    );
+    match options.listing_type.unwrap_or(ListingType::All) {
+      ListingType::Subscribed => query = query.filter(is_subscribed),
+      ListingType::Local => {
+        query = query
+          .filter(community::local.eq(true))
+          .filter(community::hidden.eq(false).or(is_subscribed));
       }
-      // If your person_id is missing, only show local
-      else {
-        match listing_type {
-          ListingType::Local => {
-            query = query
-              .filter(community::local.eq(true))
-              .filter(community::hidden.eq(false));
-          }
-          _ => query = query.filter(community::hidden.eq(false)),
-        }
+      ListingType::All => query = query.filter(community::hidden.eq(false).or(is_subscribed)),
+      ListingType::ModeratorView => {
+        query = query.filter(exists(
+          community_moderator::table.filter(
+            post::community_id
+              .eq(community_moderator::community_id)
+              .and(community_moderator::person_id.eq(person_id_join)),
+          ),
+        ));
```

Comment:
Simplified this code a bit. Behaviour is almost the same, except now you can see hidden communities that you are subscribed to even when listing_type is not specified explicitly.